api/utilities.py

```diff
@@ -181,25 +182,46 @@ def get_periods_or_range(periods: str) -> list[str]:
 
 def get_z_levels_or_range(z: str) -> list[str]:
     """
-    Function for getting the levels as a list or a range
+    Function for getting the levels filters as a list of levels, ranges
+    or range intervals
     """
     # it can be z=value1,value2,value3: z=2,10,80
     # or z=minimum value/maximum value: z=10/100
     # or z=Rn/min height/height interval: z=R20/100/50
+    # or a combination of the above: z=10,30/100,200,300,R20/100/50
+
+    values = [level_or_range.split("/") for level_or_range in split_and_strip(z)]
+
     try:
-        split_on_slash = z.split("/")
-        if len(split_on_slash) == 2:
-            z_min = convert_m_to_cm(split_on_slash[0]) if split_on_slash[0] != ".." else -sys.maxsize - 1
-            z_max = convert_m_to_cm(split_on_slash[1]) if split_on_slash[1] != ".." else sys.maxsize
-            return [f"{z_min}/{z_max}"]
-        elif len(split_on_slash) > 2:
-            return get_z_values_from_interval(split_on_slash)
-        else:
-            return [convert_m_to_cm(level) for level in z.split(",")]
+        nested_filters = [
+            (
+                get_z_values_from_interval(level_or_range)
+                if len(level_or_range) > 2
+                else (
+                    get_z_values_from_range(level_or_range)
+                    if len(level_or_range) == 2
+                    else [convert_m_to_cm(level_or_range[0])]
+                )
+            )
+            for level_or_range in values
+        ]
```

Comment:
This is way beyond my tolerance for list (in)comprehension.

Can we just call the old `get_z_levels_or_range` (the rewrite to `get_z_values_from_range` is good),
and just call it with something like:
```
values = itertools.chain(*[get_z_levels_or_range(lever_or_range) for level_or_range in split_and_strip(z)])
```
I haven't tested this, but you probably get the idea.

What do you think?

Reply:
Made the old `get_z_levels_or_range` to a nested function so it would be clearer: https://github.com/EURODEO/e-soh/pull/214/commits/f119e88e7f8435f72a3718fd7b372bd3e8fc2121